Context support

Saulis opened the pull request:
Fixes #1 
Closes #3 
Depends on #9 (rebase this branch to master after merging #9)

- vaadin-context-menu element is the default event target.
- ContextSelector traverses the light DOM of the target element.
- Context menu does not open when no context.target is available.
- Event detail is set to context.detail
- Overlay contents are now wrapped in a template do diff from the wrapped
  target elements.
- Template is stamped when overlay is opened for the first time.
- openOn is added to reflect closeOn.
- Removed listeners for `iron-select` and `iron-deselect` as they add an
  implicit dependency to iron-selectable. Also, the handler implementation
  doesn't work in Shadow DOM after switching to use template for the overlay
  contents.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/vaadin/vaadin-context-menu/11)
<!-- Reviewable:end -->

manolo commented:
Review status: 0 of 12 files reviewed at latest revision, 1 unresolved discussion.

---

*[a discussion](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11#-KRc3244RD3Xnneud2uJ:-KRc3244RD3Xnneud2uK:b-136uvg) (no related file):*
To me having a grid (for example) inside a context-menu is quite odd.
To me makes sense that the context-menu is inside the grid, by the grid, or by the body. 
I'm not discussing here whether the implementation is simpler but the readability of the declarative document.
Another thing is that the implementation does not cover the case of having multiple menus bound to the same grid, or am i wrong?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11)*
<!-- Sent from Reviewable.io -->

Saulis commented:
Review status: 0 of 12 files reviewed at latest revision, 1 unresolved discussion.

---

*[a discussion](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11#-KRc3244RD3Xnneud2uJ:-KRc7OcrDBcTWp-DboRa:b-lobz9s) (no related file):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary>
> To me having a grid (for example) inside a context-menu is quite odd.
> To me makes sense that the context-menu is inside the grid, by the grid, or by the body. 
> I'm not discussing here whether the implementation is simpler but the readability of the declarative document.
> Another thing is that the implementation does not cover the case of having multiple menus bound to the same grid, or am i wrong?
</details>
You can do all of those use cases by using the `target` property. Take a look at the `demo/advanced.html`

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11)*
<!-- Sent from Reviewable.io -->

manolo commented:
Review status: 0 of 12 files reviewed at latest revision, 1 unresolved discussion.

---

*[a discussion](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11#-KRc3244RD3Xnneud2uJ:-KRcF7ke55nJAn0Ym-8Y:b63910y) (no related file):*
<details><summary><i>Previously, Saulis (Sauli Tähkäpää) wrote…</i></summary>
> You can do all of those use cases by using the `target` property. Take a look at the `demo/advanced.html`
</details>
How about declarative target ? 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11)*
<!-- Sent from Reviewable.io -->

Saulis commented:
Review status: 0 of 12 files reviewed at latest revision, 1 unresolved discussion.

---

*[a discussion](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11#-KRc3244RD3Xnneud2uJ:-KRcNDVH0c8rjLhGMh0m:b74luqx) (no related file):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary>
> How about declarative target ? 
</details>
If you mean the old `target-selector` property, we decided to drop it out. There are two main reasons: 1) the menu or target may not be attached yet when the selector is set 2) it's not obvious which parent element to scope the `querySelector` to. There might not even be a parent element at all.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11)*
<!-- Sent from Reviewable.io -->

platosha commented:
Review status: 0 of 12 files reviewed at latest revision, 2 unresolved discussions.

---

*[vaadin-context-menu.html, line 159 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11#-KRcSX_s4Vhk-7h8gUZh:-KRcSX_t4Xzogr5xk4AQ:bw7fcae) ([raw file](https://github.com/vaadin/vaadin-context-menu/blob/b7ebc127fbe247dcc4f1ac6364a651bb2658db50/vaadin-context-menu.html#L159)):*
> ```HTML
> 
>           this.$.overlay.resetFit();
>           this.$.overlay.x = e.clientX;
> ```

Custom events dispatched by `Polymer.Base.fire` don’t have `clientX`/`clientY`. We should consider using `e.detail.x`/`e.detail.y` over `e.clientX`/`e.clientY`, in case if they are present, to support positioning with custom events.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11)*
<!-- Sent from Reviewable.io -->

platosha commented:
Reviewed 1 of 12 files at r1.
Review status: 1 of 12 files reviewed at latest revision, 5 unresolved discussions.

---

*[a discussion](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11#-KRc3244RD3Xnneud2uJ:-KRcgNbbB7YYcIOSA9d3:b-s0fst2) (no related file):*
<details><summary><i>Previously, Saulis (Sauli Tähkäpää) wrote…</i></summary>
> If you mean the old `target-selector` property, we decided to drop it out. There are two main reasons: 1) the menu or target may not be attached yet when the selector is set 2) it's not obvious which parent element to scope the `querySelector` to. There might not even be a parent element at all.
</details>
Added a couple of questions. These are primarily for a discussion, could be considered and implemented later as well.

The PR <img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

---

*[vaadin-context-menu.html, line 42 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11#-KRcapbRa5VRZ6V-4K24:-KRcapbRa5VRZ6V-4K25:b-222lqv) ([raw file](https://github.com/vaadin/vaadin-context-menu/blob/b7ebc127fbe247dcc4f1ac6364a651bb2658db50/vaadin-context-menu.html#L42)):*
> ```HTML
>          * of the element the context menu was opened on.
>          */
>         contextSelector: String,
> ```

Maybe rename to `selector`?

Read:
```html
<vaadin-context-menu selector="some-thing">
</vaadin-context-menu>
```

`contextSelector` doesn’t make much sense, since the `context` property no longer directly corresponds to the `contextSelector`.

---

*[vaadin-context-menu.html, line 142 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11#-KRcbGdOMwB3rPBB3tYT:-KRcbGdOMwB3rPBB3tYU:b-s63czt) ([raw file](https://github.com/vaadin/vaadin-context-menu/blob/b7ebc127fbe247dcc4f1ac6364a651bb2658db50/vaadin-context-menu.html#L142)):*
> ```HTML
>       _contextTarget: function(e) {
>         if (this.contextSelector) {
>           var target = Polymer.dom(this.target).querySelector(this.contextSelector);
> ```

This probably doesn’t work for multiple targets. Consider:

```html
<vaadin-context-menu context-selector="div">
  <div>First target</div>
  <p>Not target</p>
  <div>Second target</div>
</vaadin-context-menu>
```

Should we also make the second target work?

---

*[vaadin-context-menu.html, line 157 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11#-KRcfc8dLVJeVaJqiiGi:-KRcfc8dLVJeVaJqiiGj:b-xq4rmx) ([raw file](https://github.com/vaadin/vaadin-context-menu/blob/b7ebc127fbe247dcc4f1ac6364a651bb2658db50/vaadin-context-menu.html#L157)):*
> ```HTML
>         if (this.context.target) {
>           e.preventDefault();
> ```

Would it make sense to assign `context.detail` and `context.target` as first-level properties for the `this._instance`?

Consider to support:

```html
<vaadin-context-menu>
  <template>
    <paper-menu>
      <paper-item>[[target.localName]] [[detail.prop]]</paper-item>
    </paper-menu>
  </template>
</vaadin-context-menu>
```

vs

```html
<vaadin-context-menu>
  <template>
    <paper-menu>
      <paper-item>[[context.target.localName]] [[context.detail.prop]]</paper-item>
    </paper-menu>
  </template>
</vaadin-context-menu>
```

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11)*
<!-- Sent from Reviewable.io -->

Saulis commented:
Review status: 1 of 12 files reviewed at latest revision, 5 unresolved discussions.

---

*[vaadin-context-menu.html, line 42 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11#-KRcapbRa5VRZ6V-4K24:-KRciIOyEcHGdpVOe44m:b2nhswy) ([raw file](https://github.com/vaadin/vaadin-context-menu/blob/b7ebc127fbe247dcc4f1ac6364a651bb2658db50/vaadin-context-menu.html#L42)):*
<details><summary><i>Previously, platosha (Anton Platonov) wrote…</i></summary>
> Maybe rename to `selector`?
> 
> Read:
> ```html
> <vaadin-context-menu selector="some-thing">
> </vaadin-context-menu>
> ```
> 
> `contextSelector` doesn’t make much sense, since the `context` property no longer directly corresponds to the `contextSelector`.
</details>
Yes

---

*[vaadin-context-menu.html, line 142 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11#-KRcbGdOMwB3rPBB3tYT:-KRciGxTt1rULMXAi29b:b2nhswy) ([raw file](https://github.com/vaadin/vaadin-context-menu/blob/b7ebc127fbe247dcc4f1ac6364a651bb2658db50/vaadin-context-menu.html#L142)):*
<details><summary><i>Previously, platosha (Anton Platonov) wrote…</i></summary>
> This probably doesn’t work for multiple targets. Consider:
> 
> ```html
> <vaadin-context-menu context-selector="div">
>   <div>First target</div>
>   <p>Not target</p>
>   <div>Second target</div>
> </vaadin-context-menu>
> ```
> 
> Should we also make the second target work?
</details>
Yes

---

*[vaadin-context-menu.html, line 157 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11#-KRcfc8dLVJeVaJqiiGi:-KRcj68SJ1INwIQwgzv3:b-m15fwb) ([raw file](https://github.com/vaadin/vaadin-context-menu/blob/b7ebc127fbe247dcc4f1ac6364a651bb2658db50/vaadin-context-menu.html#L157)):*
<details><summary><i>Previously, platosha (Anton Platonov) wrote…</i></summary>
> Would it make sense to assign `context.detail` and `context.target` as first-level properties for the `this._instance`?
> 
> Consider to support:
> 
> ```html
> <vaadin-context-menu>
>   <template>
>     <paper-menu>
>       <paper-item>[[target.localName]] [[detail.prop]]</paper-item>
>     </paper-menu>
>   </template>
> </vaadin-context-menu>
> ```
> 
> vs
> 
> ```html
> <vaadin-context-menu>
>   <template>
>     <paper-menu>
>       <paper-item>[[context.target.localName]] [[context.detail.prop]]</paper-item>
>     </paper-menu>
>   </template>
> </vaadin-context-menu>
> ```
</details>
Yes. Let's drop `context`. To reduce confusion, let's rename `target` property on `vaadin-context-menu` to something else.

---

*[vaadin-context-menu.html, line 159 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11#-KRcSX_s4Vhk-7h8gUZh:-KRcj9iYMiM1z2f0zOOh:bb77ufg) ([raw file](https://github.com/vaadin/vaadin-context-menu/blob/b7ebc127fbe247dcc4f1ac6364a651bb2658db50/vaadin-context-menu.html#L159)):*
<details><summary><i>Previously, platosha (Anton Platonov) wrote…</i></summary>
> Custom events dispatched by `Polymer.Base.fire` don’t have `clientX`/`clientY`. We should consider using `e.detail.x`/`e.detail.y` over `e.clientX`/`e.clientY`, in case if they are present, to support positioning with custom events.
</details>
Yes.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11)*
<!-- Sent from Reviewable.io -->

manolo commented:
Review status: 1 of 12 files reviewed at latest revision, 7 unresolved discussions, some commit checks failed.

---

*[vaadin-context-menu-overlay.html, line 21 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11#-KRhFRZJIj1WHFzhhH0N:-KRhFRZJIj1WHFzhhH0O:bhnr20i) ([raw file](https://github.com/vaadin/vaadin-context-menu/blob/b7ebc127fbe247dcc4f1ac6364a651bb2658db50/vaadin-context-menu-overlay.html#L21)):*
> ```HTML
>         min-height: 32px !important;
>         padding: 0 24px;
>       }
> ```

Remove all these styles as discussed.

---

*[vaadin-context-menu.html, line 91 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11#-KRhFviBz3w86luzE2wh:-KRhFviBz3w86luzE2wi:b-ocgjbr) ([raw file](https://github.com/vaadin/vaadin-context-menu/blob/b7ebc127fbe247dcc4f1ac6364a651bb2658db50/vaadin-context-menu.html#L91)):*
> ```HTML
> 
>       _onOverlayOpened: function() {
>         var child = Polymer.dom(this.$.content).querySelector('*');
> ```

Couldn't the user put in the content be anything, not just  a child. And the first child could not be focusable.
```
<div> caption</div>
<paper-menu>...</paper-menu>
```

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11)*
<!-- Sent from Reviewable.io -->

manolo commented:
Review status: 1 of 12 files reviewed at latest revision, 8 unresolved discussions, some commit checks failed.

---

*[docs/vaadin-context-menu.adoc, line 10 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11#-KRhHTIURsuN4_I8nv81:-KRhHTIVga6d-6Ft4-XR:b-ocffuk) ([raw file](https://github.com/vaadin/vaadin-context-menu/blob/b7ebc127fbe247dcc4f1ac6364a651bb2658db50/docs/vaadin-context-menu.adoc#L10)):*
> ```AsciiDoc
> = Overview
> 
> [vaadinelement]#vaadin-context-menu# is a Polymer element implementing a menu in an overlay which is open by a context click.
> ```

I would omit the action to open it. In fact it would be opened with long tap, or even be used in a classic app menu bar where click or hovering is used normally.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11)*
<!-- Sent from Reviewable.io -->

manolo commented:
Review status: 1 of 12 files reviewed at latest revision, 12 unresolved discussions, some commit checks failed.

---

*[docs/vaadin-context-menu.adoc, line 21 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11#-KRhIQiMNF743LOCggnO:-KRhIQiMNF743LOCggnP:b-hb7ja2) ([raw file](https://github.com/vaadin/vaadin-context-menu/blob/b7ebc127fbe247dcc4f1ac6364a651bb2658db50/docs/vaadin-context-menu.adoc#L21)):*
> ```AsciiDoc
> [vaadinelement]#vaadin-context-menu# uses a [propertyname]#template# element inside to determine the contents for the overlay.
> 
> By default, [vaadinelement]#vaadin-context-menu# itself is used as a target to listen for [propertyname]#contextmenu# events.
> ```

... as the target to listen to ...

---

*[docs/vaadin-context-menu.adoc, line 22 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11#-KRhIYxPcEh1Eu6cBiWY:-KRhIYxQ3L77f0yUklw8:bpmyji9) ([raw file](https://github.com/vaadin/vaadin-context-menu/blob/b7ebc127fbe247dcc4f1ac6364a651bb2658db50/docs/vaadin-context-menu.adoc#L22)):*
> ```AsciiDoc
> 
> By default, [vaadinelement]#vaadin-context-menu# itself is used as a target to listen for [propertyname]#contextmenu# events.
> In practice, any child element of the target element can be context clicked.
> ```

... target element will respond to [propertyname]#contextmenu# events for opening the menu. The event type can be customised by the user.

---

*[docs/vaadin-context-menu.adoc, line 51 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11#-KRhJG6FVJ55a9dcJWAd:-KRhJG6FVJ55a9dcJWAe:b-ylich8) ([raw file](https://github.com/vaadin/vaadin-context-menu/blob/b7ebc127fbe247dcc4f1ac6364a651bb2658db50/docs/vaadin-context-menu.adoc#L51)):*
> ```AsciiDoc
> 
> In case you don't want [vaadinelement]#vaadin-context-menu# as your target element for clicks, you can target any element outside by
> setting the [propertyname]#target# value.
> ```

... by imperatively setting ... element in your code.

---

*[docs/vaadin-context-menu.adoc, line 96 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11#-KRhJouRwtuUB3IqnRKE:-KRhJouRwtuUB3IqnRKF:b46a44b) ([raw file](https://github.com/vaadin/vaadin-context-menu/blob/b7ebc127fbe247dcc4f1ac6364a651bb2658db50/docs/vaadin-context-menu.adoc#L96)):*
> ```AsciiDoc
> By default, the overlay opens when the [propertyname]#target# element receives a [propertyname]#contextmenu# event.
> 
> To replace [propertyname]#contextmenu# by any other event, use [propertyname]#openOn#.
> ```

To me contextmenu property sounds redundant, why not ? 

To replace it by ... 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11)*
<!-- Sent from Reviewable.io -->

manolo commented:
Review status: 1 of 12 files reviewed at latest revision, 18 unresolved discussions, some commit checks failed.

---

*[docs/vaadin-context-menu.adoc, line 110 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11#-KRhKtYkD5dbFMJR-noT:-KRhKtYkD5dbFMJR-noU:bt8nh3g) ([raw file](https://github.com/vaadin/vaadin-context-menu/blob/b7ebc127fbe247dcc4f1ac6364a651bb2658db50/docs/vaadin-context-menu.adoc#L110)):*
> ```AsciiDoc
> ----
> 
> Overlay can be also opened programmatically using [propertyname]#opened#.
> ```

programatically setting the property ...

---

*[docs/vaadin-context-menu.adoc, line 130 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11#-KRhL35TF3lDlRBsoPlW:-KRhL35TF3lDlRBsoPlX:b-9ou7no) ([raw file](https://github.com/vaadin/vaadin-context-menu/blob/b7ebc127fbe247dcc4f1ac6364a651bb2658db50/docs/vaadin-context-menu.adoc#L130)):*
> ```AsciiDoc
> = Context
> 
> Whenever the overlay opens, the context is set to [propertyname]#context# property.
> ```

a context object is assigned to the context property of the element.

---

*[docs/vaadin-context-menu.adoc, line 131 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11#-KRhLVW0bzlMuqJUkAig:-KRhLVW0bzlMuqJUkAih:bij30gf) ([raw file](https://github.com/vaadin/vaadin-context-menu/blob/b7ebc127fbe247dcc4f1ac6364a651bb2658db50/docs/vaadin-context-menu.adoc#L131)):*
> ```AsciiDoc
> 
> Whenever the overlay opens, the context is set to [propertyname]#context# property.
> The context has two properties, [propertyname]#detail# and [propertyname]#target#.
> ```

The context object has two properties: 

---

*[docs/vaadin-context-menu.adoc, line 132 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11#-KRhM1AS1gnwiE26SmU3:-KRhM1ATgZ-Hr0ABKOSW:b-oscase) ([raw file](https://github.com/vaadin/vaadin-context-menu/blob/b7ebc127fbe247dcc4f1ac6364a651bb2658db50/docs/vaadin-context-menu.adoc#L132)):*
> ```AsciiDoc
> Whenever the overlay opens, the context is set to [propertyname]#context# property.
> The context has two properties, [propertyname]#detail# and [propertyname]#target#.
> They are especially helpful when defining dynamic content for the overlay based on the context.
> ```

... when showing different content in the overlay based on the UI element that opens it.

---

*[docs/vaadin-context-menu.adoc, line 134 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11#-KRhMw5waddW4_MrjW5p:-KRhMw5x1vd36ibp3guC:b55qdbz) ([raw file](https://github.com/vaadin/vaadin-context-menu/blob/b7ebc127fbe247dcc4f1ac6364a651bb2658db50/docs/vaadin-context-menu.adoc#L134)):*
> ```AsciiDoc
> They are especially helpful when defining dynamic content for the overlay based on the context.
> 
> [propertyname]#context.target# is the element from which the event originated from.
> ```

The context.target is the element that originated the event.

---

*[docs/vaadin-context-menu.adoc, line 148 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11#-KRhN5C0_Wo2z6z4REwZ:-KRhN5C0_Wo2z6z4REw_:bald5ml) ([raw file](https://github.com/vaadin/vaadin-context-menu/blob/b7ebc127fbe247dcc4f1ac6364a651bb2658db50/docs/vaadin-context-menu.adoc#L148)):*
> ```AsciiDoc
> ----
> 
> [propertyname]#context.detail# contains the event details from the originating event, which
> ```

The context.details has the event details of the original event. That is useful when the event is fired inside custom elements.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11)*
<!-- Sent from Reviewable.io -->

tomivirkki commented:
*[vaadin-context-menu.html, line 129 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11#-KRhoSkmRvMhqrIelhQJ:-KRhoSkn2IkOafLe4cT3:b-j016ze) ([raw file](https://github.com/vaadin/vaadin-context-menu/blob/b7ebc127fbe247dcc4f1ac6364a651bb2658db50/vaadin-context-menu.html#L129)):*
> ```HTML
>       _openedChanged: function(opened) {
>         if (opened) {
>           this._contentTemplate = Polymer.dom(this).querySelector('template');
> ```

'template' might be too vague a selector now that there can be arbitrary content (even a template?) next to it as well. Could have a class marker or something?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11)*
<!-- Sent from Reviewable.io -->

tomivirkki commented:
*[vaadin-context-menu.html, line 182 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11#-KRhu5mCPI3Re7b0g0Ha:-KRhu5mCPI3Re7b0g0Hb:b9edjgh) ([raw file](https://github.com/vaadin/vaadin-context-menu/blob/b7ebc127fbe247dcc4f1ac6364a651bb2658db50/vaadin-context-menu.html#L182)):*
> ```HTML
> 
>           this._instance = this.stamp({});
>           var content = this.$.overlay.querySelector('.dropdown-content');
> ```

Unused variable

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11)*
<!-- Sent from Reviewable.io -->

Saulis commented:
Review status: 1 of 12 files reviewed at latest revision, 20 unresolved discussions, some commit checks failed.

---

*[vaadin-context-menu-overlay.html, line 21 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11#-KRhFRZJIj1WHFzhhH0N:-KRlsjTqGJrNu8fntVlf:b-896fix) ([raw file](https://github.com/vaadin/vaadin-context-menu/blob/b7ebc127fbe247dcc4f1ac6364a651bb2658db50/vaadin-context-menu-overlay.html#L21)):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary>
> Remove all these styles as discussed.
</details>
Done.

---

*[vaadin-context-menu.html, line 42 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11#-KRcapbRa5VRZ6V-4K24:-KRlu0RjrZabDiMw00iF:b-896fix) ([raw file](https://github.com/vaadin/vaadin-context-menu/blob/b7ebc127fbe247dcc4f1ac6364a651bb2658db50/vaadin-context-menu.html#L42)):*
<details><summary><i>Previously, Saulis (Sauli Tähkäpää) wrote…</i></summary>
> Yes
</details>
Done.

---

*[vaadin-context-menu.html, line 91 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11#-KRhFviBz3w86luzE2wh:-KRluA_ssaNzv25dUP5q:bla7ouz) ([raw file](https://github.com/vaadin/vaadin-context-menu/blob/b7ebc127fbe247dcc4f1ac6364a651bb2658db50/vaadin-context-menu.html#L91)):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary>
> Couldn't the user put in the content be anything, not just  a child. And the first child could not be focusable.
> ```
> <div> caption</div>
> <paper-menu>...</paper-menu>
> ```
> 
</details>
Yes, that can happen. As this PR doesn't change the behaviour on that feature, we should open another PR for it if it's an issue.

---

*[vaadin-context-menu.html, line 129 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11#-KRhoSkmRvMhqrIelhQJ:-KRlubPaW2lkg1ZwYoZh:bqosc0x) ([raw file](https://github.com/vaadin/vaadin-context-menu/blob/b7ebc127fbe247dcc4f1ac6364a651bb2658db50/vaadin-context-menu.html#L129)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary>
> 'template' might be too vague a selector now that there can be arbitrary content (even a template?) next to it as well. Could have a class marker or something?
</details>
Well.. not sure. I was thinking it should be enough to document that the first `<template>` element is used for the overlay contents? Not opposing adding classes if that's better.

---

*[vaadin-context-menu.html, line 142 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11#-KRcbGdOMwB3rPBB3tYT:-KRlzNgqaaTt6GNy0UBn:b-896fix) ([raw file](https://github.com/vaadin/vaadin-context-menu/blob/b7ebc127fbe247dcc4f1ac6364a651bb2658db50/vaadin-context-menu.html#L142)):*
<details><summary><i>Previously, Saulis (Sauli Tähkäpää) wrote…</i></summary>
> Yes
</details>
Done.

---

*[vaadin-context-menu.html, line 157 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11#-KRcfc8dLVJeVaJqiiGi:-KRm5QspykDu0DDFSoGI:bb7ierp) ([raw file](https://github.com/vaadin/vaadin-context-menu/blob/b7ebc127fbe247dcc4f1ac6364a651bb2658db50/vaadin-context-menu.html#L157)):*
<details><summary><i>Previously, Saulis (Sauli Tähkäpää) wrote…</i></summary>
> Yes. Let's drop `context`. To reduce confusion, let's rename `target` property on `vaadin-context-menu` to something else.
</details>
Done. Renamed `target` to `listenOn` we can discuss alternatives.

---

*[vaadin-context-menu.html, line 159 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11#-KRcSX_s4Vhk-7h8gUZh:-KRlzy-t8uHA7O09Qmsd:b-896fix) ([raw file](https://github.com/vaadin/vaadin-context-menu/blob/b7ebc127fbe247dcc4f1ac6364a651bb2658db50/vaadin-context-menu.html#L159)):*
<details><summary><i>Previously, Saulis (Sauli Tähkäpää) wrote…</i></summary>
> Yes.
</details>
Done.

---

*[vaadin-context-menu.html, line 182 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11#-KRhu5mCPI3Re7b0g0Ha:-KRm-1pR2bFdaBXrhTCh:bss4a2b) ([raw file](https://github.com/vaadin/vaadin-context-menu/blob/b7ebc127fbe247dcc4f1ac6364a651bb2658db50/vaadin-context-menu.html#L182)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary>
> Unused variable
</details>
Removed.

---

*[docs/vaadin-context-menu.adoc, line 10 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11#-KRhHTIURsuN4_I8nv81:-KRm_dfoO_ONMj3amsQy:bw41l53) ([raw file](https://github.com/vaadin/vaadin-context-menu/blob/b7ebc127fbe247dcc4f1ac6364a651bb2658db50/docs/vaadin-context-menu.adoc#L10)):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary>
> I would omit the action to open it. In fact it would be opened with long tap, or even be used in a classic app menu bar where click or hovering is used normally.
</details>
Hmm.. I was thinking "context click" is a commonly understood concept that also translates into touch devices. The way how the overlay is opened is kinda the feature that makes it a "context" menu, don't you think?

---

*[docs/vaadin-context-menu.adoc, line 21 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11#-KRhIQiMNF743LOCggnO:-KRmarSyg3Y-p9G_nnzk:b-896fix) ([raw file](https://github.com/vaadin/vaadin-context-menu/blob/b7ebc127fbe247dcc4f1ac6364a651bb2658db50/docs/vaadin-context-menu.adoc#L21)):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary>
> ... as the target to listen to ...
</details>
Done.

---

*[docs/vaadin-context-menu.adoc, line 22 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11#-KRhIYxPcEh1Eu6cBiWY:-KRmd7cEMg_1ynCNZdtC:b-896fix) ([raw file](https://github.com/vaadin/vaadin-context-menu/blob/b7ebc127fbe247dcc4f1ac6364a651bb2658db50/docs/vaadin-context-menu.adoc#L22)):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary>
> ... target element will respond to [propertyname]#contextmenu# events for opening the menu. The event type can be customised by the user.
</details>
Done.

---

*[docs/vaadin-context-menu.adoc, line 51 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11#-KRhJG6FVJ55a9dcJWAd:-KRmdE_XRL1cUkQiCEtl:b-9xnbav) ([raw file](https://github.com/vaadin/vaadin-context-menu/blob/b7ebc127fbe247dcc4f1ac6364a651bb2658db50/docs/vaadin-context-menu.adoc#L51)):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary>
> ... by imperatively setting ... element in your code.
</details>
Not sure if it's necessary to use so explicit wording – to be precise, the `listenOn` can also be set using two-way bindings declaratively if a another element exposes a reference as a property. I've used that sometimes.

---

*[docs/vaadin-context-menu.adoc, line 96 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11#-KRhJouRwtuUB3IqnRKE:-KRmdkF7Iz6wQAZ5of0i:b-896fix) ([raw file](https://github.com/vaadin/vaadin-context-menu/blob/b7ebc127fbe247dcc4f1ac6364a651bb2658db50/docs/vaadin-context-menu.adoc#L96)):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary>
> To me contextmenu property sounds redundant, why not ? 
> 
> To replace it by ... 
</details>
Done.

---

*[docs/vaadin-context-menu.adoc, line 110 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11#-KRhKtYkD5dbFMJR-noT:-KRmdv6SlycXONWEd5PK:b-896fix) ([raw file](https://github.com/vaadin/vaadin-context-menu/blob/b7ebc127fbe247dcc4f1ac6364a651bb2658db50/docs/vaadin-context-menu.adoc#L110)):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary>
> programatically setting the property ...
</details>
Done.

---

*[docs/vaadin-context-menu.adoc, line 134 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11#-KRhMw5waddW4_MrjW5p:-KRmexFyMbAu7YcWyUAL:b-896fix) ([raw file](https://github.com/vaadin/vaadin-context-menu/blob/b7ebc127fbe247dcc4f1ac6364a651bb2658db50/docs/vaadin-context-menu.adoc#L134)):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary>
> The context.target is the element that originated the event.
</details>
Done.

---

*[docs/vaadin-context-menu.adoc, line 148 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11#-KRhN5C0_Wo2z6z4REwZ:-KRmewj3jk4rL0aVzHIk:b-896fix) ([raw file](https://github.com/vaadin/vaadin-context-menu/blob/b7ebc127fbe247dcc4f1ac6364a651bb2658db50/docs/vaadin-context-menu.adoc#L148)):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary>
> The context.details has the event details of the original event. That is useful when the event is fired inside custom elements.
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11)*
<!-- Sent from Reviewable.io -->

tomivirkki commented:
*[vaadin-context-menu.html, line 129 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11#-KRhoSkmRvMhqrIelhQJ:-KRmoUUnxh--iXcNwnLK:b-za4pi4) ([raw file](https://github.com/vaadin/vaadin-context-menu/blob/b7ebc127fbe247dcc4f1ac6364a651bb2658db50/vaadin-context-menu.html#L129)):*
<details><summary><i>Previously, Saulis (Sauli Tähkäpää) wrote…</i></summary>
> Well.. not sure. I was thinking it should be enough to document that the first `<template>` element is used for the overlay contents? Not opposing adding classes if that's better.
</details>
Yeah can't really say which one is better...

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11)*
<!-- Sent from Reviewable.io -->

platosha commented:
Reviewed 1 of 12 files at r1, 3 of 9 files at r2.
Review status: 4 of 11 files reviewed at latest revision, 18 unresolved discussions, some commit checks failed.

---

*[vaadin-context-menu.html, line 129 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11#-KRhoSkmRvMhqrIelhQJ:-KRmng4tOnN_Z7RsWA5z:b32bx1o) ([raw file](https://github.com/vaadin/vaadin-context-menu/blob/b7ebc127fbe247dcc4f1ac6364a651bb2658db50/vaadin-context-menu.html#L129)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary>
> Yeah can't really say which one is better...
</details>
How about searching only on the immediate children of the context menu instead? This way the only `<template>` corresponding to the menu will be picked up, no matter if it was first or not.

---

*[vaadin-context-menu.html, line 159 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11#-KRcSX_s4Vhk-7h8gUZh:-KRmp6gSG23d71djwKH2:b7a2xg7) ([raw file](https://github.com/vaadin/vaadin-context-menu/blob/b7ebc127fbe247dcc4f1ac6364a651bb2658db50/vaadin-context-menu.html#L159)):*
<details><summary><i>Previously, Saulis (Sauli Tähkäpää) wrote…</i></summary>
> Done.
</details>
I thought it will be better the other way around, check the detail values first.

In case if a custom event is a mouse event, it might be useful to prioritise the detail values over the event values to make it easier to override.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11)*
<!-- Sent from Reviewable.io -->

Saulis commented:
Review status: 3 of 11 files reviewed at latest revision, 18 unresolved discussions.

---

*[vaadin-context-menu.html, line 129 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11#-KRhoSkmRvMhqrIelhQJ:-KRmwRbfZOejDtaDJFP7:b-896fix) ([raw file](https://github.com/vaadin/vaadin-context-menu/blob/b7ebc127fbe247dcc4f1ac6364a651bb2658db50/vaadin-context-menu.html#L129)):*
<details><summary><i>Previously, platosha (Anton Platonov) wrote…</i></summary>
> How about searching only on the immediate children of the context menu instead? This way the only `<template>` corresponding to the menu will be picked up, no matter if it was first or not.
</details>
Done.

---

*[vaadin-context-menu.html, line 159 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11#-KRcSX_s4Vhk-7h8gUZh:-KRmxRKUFzCPF7oHSes0:b-896fix) ([raw file](https://github.com/vaadin/vaadin-context-menu/blob/b7ebc127fbe247dcc4f1ac6364a651bb2658db50/vaadin-context-menu.html#L159)):*
<details><summary><i>Previously, platosha (Anton Platonov) wrote…</i></summary>
> I thought it will be better the other way around, check the detail values first.
> 
> In case if a custom event is a mouse event, it might be useful to prioritise the detail values over the event values to make it easier to override.
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11)*
<!-- Sent from Reviewable.io -->

manolo commented:
Review status: 3 of 11 files reviewed at latest revision, 10 unresolved discussions, some commit checks failed.

---

*[docs/vaadin-context-menu.adoc, line 10 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11#-KRhHTIURsuN4_I8nv81:-KRnqAQi3lUP7AIVieyD:b-kmk2lb) ([raw file](https://github.com/vaadin/vaadin-context-menu/blob/b7ebc127fbe247dcc4f1ac6364a651bb2658db50/docs/vaadin-context-menu.adoc#L10)):*
<details><summary><i>Previously, Saulis (Sauli Tähkäpää) wrote…</i></summary>
> Hmm.. I was thinking "context click" is a commonly understood concept that also translates into touch devices. The way how the overlay is opened is kinda the feature that makes it a "context" menu, don't you think?
</details>
So I would add the default word somewhere, maybe:

... overlay that by default is opened when the user right-clicks on a UI element. 

---

*[docs/vaadin-context-menu.adoc, line 51 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11#-KRhJG6FVJ55a9dcJWAd:-KRnrD2d2RKWxR8TvdxB:b-29xta7) ([raw file](https://github.com/vaadin/vaadin-context-menu/blob/b7ebc127fbe247dcc4f1ac6364a651bb2658db50/docs/vaadin-context-menu.adoc#L51)):*
<details><summary><i>Previously, Saulis (Sauli Tähkäpää) wrote…</i></summary>
> Not sure if it's necessary to use so explicit wording – to be precise, the `listenOn` can also be set using two-way bindings declaratively if a another element exposes a reference as a property. I've used that sometimes.
</details>
But that element should be set imperatively somehow, I mean you can have `listen-on="_mySharedElement"` but _mySharedElement should be computed in code

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11)*
<!-- Sent from Reviewable.io -->

Saulis commented:
Review status: 3 of 11 files reviewed at latest revision, 10 unresolved discussions.

---

*[docs/vaadin-context-menu.adoc, line 10 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11#-KRhHTIURsuN4_I8nv81:-KS09klViX9uORCnJZCX:b-jnqdve) ([raw file](https://github.com/vaadin/vaadin-context-menu/blob/b7ebc127fbe247dcc4f1ac6364a651bb2658db50/docs/vaadin-context-menu.adoc#L10)):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary>
> So I would add the default word somewhere, maybe:
> 
> ... overlay that by default is opened when the user right-clicks on a UI element. 
</details>
Right-click doesn't translate to touch devices. Should we use the word "gesture" or something that translates to both desktop and touch devices instead?

---

*[docs/vaadin-context-menu.adoc, line 51 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11#-KRhJG6FVJ55a9dcJWAd:-KS0A_MUTXJCsw_ICN86:b478o9y) ([raw file](https://github.com/vaadin/vaadin-context-menu/blob/b7ebc127fbe247dcc4f1ac6364a651bb2658db50/docs/vaadin-context-menu.adoc#L51)):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary>
> But that element should be set imperatively somehow, I mean you can have `listen-on="_mySharedElement"` but _mySharedElement should be computed in code
</details>
That's right. I'm probably just nitpicking here, don't have a strong opinion about the wording.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11)*
<!-- Sent from Reviewable.io -->

platosha commented:
Review status: 3 of 11 files reviewed at latest revision, 10 unresolved discussions.

---

*[vaadin-context-menu.html, line 152 at r3](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11#-KS0CaRuwJG5dExfSrzn:-KS0CaRuwJG5dExfSrzo:bx9g3wp) ([raw file](https://github.com/vaadin/vaadin-context-menu/blob/37fbe2c26b65d97c7bda2ba8a1ea86c4eb371b23/vaadin-context-menu.html#L152)):*
> ```HTML
>         if (opened) {
>           this._contentTemplate = Polymer.dom(this).children.filter(function(el) {
>             return el.tagName === 'TEMPLATE';
> ```

Maybe use the lowercase `localName`?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11)*
<!-- Sent from Reviewable.io -->

tehapo commented:
Reviewed 3 of 12 files at r1, 6 of 9 files at r2, 2 of 2 files at r3.
Review status: all files reviewed at latest revision, 14 unresolved discussions.

---

*[vaadin-context-menu.html, line 113 at r3](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11#-KS0AmHNJ_eloR64dScw:-KS0AmHNJ_eloR64dScx:beecxra) ([raw file](https://github.com/vaadin/vaadin-context-menu/blob/37fbe2c26b65d97c7bda2ba8a1ea86c4eb371b23/vaadin-context-menu.html#L113)):*
> ```HTML
> 
>       _onOverlayOpened: function() {
>         var child = Polymer.dom(this.$.content).querySelector('*');
> ```

Using `Polymer.dom(this.$.content).firstElementChild` might be a bit cleaner here?

---

*[demo/index.html, line 15 at r3](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11#-KS0BCMfPMFE1mBvxG6R:-KS0BCMfPMFE1mBvxG6S:b-phwho2) ([raw file](https://github.com/vaadin/vaadin-context-menu/blob/37fbe2c26b65d97c7bda2ba8a1ea86c4eb371b23/demo/index.html#L15)):*
> ```HTML
>   <link rel="import" href="common.html">
>   <link rel="import" href="../../paper-button/paper-button.html">
>   <link rel="import" href="../../paper-input/paper-input.html">
> ```

These imports are not used anywhere?

---

*[demo/index.html, line 37 at r3](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11#-KS0HflU617GOJwYMis6:-KS0HflVKQhOBagdeddw:bdm5kfw) ([raw file](https://github.com/vaadin/vaadin-context-menu/blob/37fbe2c26b65d97c7bda2ba8a1ea86c4eb371b23/demo/index.html#L37)):*
> ```HTML
>     <demo-snippet>
>       <style is="custom-style">
> ```

Couldn't this `style` tag be actually removed instead of just adding a line break there? :)

---

*[docs/vaadin-context-menu.adoc, line 13 at r3](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11#-KS0G1kDHkANB4uNPJLw:-KS0G1kEWw7_aWAuwFQX:b-ucr7dk) ([raw file](https://github.com/vaadin/vaadin-context-menu/blob/37fbe2c26b65d97c7bda2ba8a1ea86c4eb371b23/docs/vaadin-context-menu.adoc#L13)):*
> ```AsciiDoc
> 
> [[figure.vaadin-context-menu.overview]]
> .A screenshot of [vaadinelement]#vaadin-context-menu#
> ```

@magi42 has said before:

"Avoid phrases like "A screenshot of" in image titles, as it's rather implicit that it's a screenshot. (Compare with a newspaper article with an image and title: "A photograph of Sauli Niinistö.")"

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11)*
<!-- Sent from Reviewable.io -->

Saulis commented:
Review status: 8 of 11 files reviewed at latest revision, 12 unresolved discussions.

---

*[vaadin-context-menu.html, line 113 at r3](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11#-KS0AmHNJ_eloR64dScw:-KS0SUyExriIOYzqhOib:b-896fix) ([raw file](https://github.com/vaadin/vaadin-context-menu/blob/37fbe2c26b65d97c7bda2ba8a1ea86c4eb371b23/vaadin-context-menu.html#L113)):*
<details><summary><i>Previously, tehapo (Teemu Pöntelin) wrote…</i></summary>
> Using `Polymer.dom(this.$.content).firstElementChild` might be a bit cleaner here?
</details>
Done.

---

*[vaadin-context-menu.html, line 152 at r3](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11#-KS0CaRuwJG5dExfSrzn:-KS0SR79Ez36Zgf7Tlst:b-896fix) ([raw file](https://github.com/vaadin/vaadin-context-menu/blob/37fbe2c26b65d97c7bda2ba8a1ea86c4eb371b23/vaadin-context-menu.html#L152)):*
<details><summary><i>Previously, platosha (Anton Platonov) wrote…</i></summary>
> Maybe use the lowercase `localName`?
</details>
Done.

---

*[demo/index.html, line 15 at r3](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11#-KS0BCMfPMFE1mBvxG6R:-KS0SEoTDAKS-WT0gmVW:b-896fix) ([raw file](https://github.com/vaadin/vaadin-context-menu/blob/37fbe2c26b65d97c7bda2ba8a1ea86c4eb371b23/demo/index.html#L15)):*
<details><summary><i>Previously, tehapo (Teemu Pöntelin) wrote…</i></summary>
> These imports are not used anywhere?
</details>
Done.

---

*[demo/index.html, line 37 at r3](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11#-KS0HflU617GOJwYMis6:-KS0SA9xRPeKQSlMNC-O:b-896fix) ([raw file](https://github.com/vaadin/vaadin-context-menu/blob/37fbe2c26b65d97c7bda2ba8a1ea86c4eb371b23/demo/index.html#L37)):*
<details><summary><i>Previously, tehapo (Teemu Pöntelin) wrote…</i></summary>
> Couldn't this `style` tag be actually removed instead of just adding a line break there? :)
</details>
Done.

---

*[docs/vaadin-context-menu.adoc, line 13 at r3](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11#-KS0G1kDHkANB4uNPJLw:-KS0S6yY7lMYLd9jihA_:b-896fix) ([raw file](https://github.com/vaadin/vaadin-context-menu/blob/37fbe2c26b65d97c7bda2ba8a1ea86c4eb371b23/docs/vaadin-context-menu.adoc#L13)):*
<details><summary><i>Previously, tehapo (Teemu Pöntelin) wrote…</i></summary>
> @magi42 has said before:
> 
> "Avoid phrases like "A screenshot of" in image titles, as it's rather implicit that it's a screenshot. (Compare with a newspaper article with an image and title: "A photograph of Sauli Niinistö.")"
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11)*
<!-- Sent from Reviewable.io -->

tehapo commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 3 of 3 files at r4.
Review status: all files reviewed at latest revision, 8 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11#-:-KS0VWXwiGO6iZB-tT8Y:bnfp4nl)*
<!-- Sent from Reviewable.io -->

manolo commented:
Review status: all files reviewed at latest revision, 8 unresolved discussions.

---

*[docs/vaadin-context-menu.adoc, line 10 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11#-KRhHTIURsuN4_I8nv81:-KS0wYmeLwyqSopf5tao:b-i32w73) ([raw file](https://github.com/vaadin/vaadin-context-menu/blob/b7ebc127fbe247dcc4f1ac6364a651bb2658db50/docs/vaadin-context-menu.adoc#L10)):*
<details><summary><i>Previously, Saulis (Sauli Tähkäpää) wrote…</i></summary>
> Right-click doesn't translate to touch devices. Should we use the word "gesture" or something that translates to both desktop and touch devices instead?
</details>
Maybe an user action ? that includes mouse, keyboard, touch ...

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11)*
<!-- Sent from Reviewable.io -->

tehapo commented:
Review status: all files reviewed at latest revision, 8 unresolved discussions.

---

*[docs/vaadin-context-menu.adoc, line 10 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11#-KRhHTIURsuN4_I8nv81:-KS1A9prPnZse-jqaGz5:b-4ekahg) ([raw file](https://github.com/vaadin/vaadin-context-menu/blob/b7ebc127fbe247dcc4f1ac6364a651bb2658db50/docs/vaadin-context-menu.adoc#L10)):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary>
> Maybe an user action ? that includes mouse, keyboard, touch ...
</details>
Why not simply mention both options – "which by default is opened by a context click or a long press touch event"?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11)*
<!-- Sent from Reviewable.io -->

manolo commented:
Review status: all files reviewed at latest revision, 8 unresolved discussions.

---

*[docs/vaadin-context-menu.adoc, line 10 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11#-KRhHTIURsuN4_I8nv81:-KS57SBPGPm3v6GUkYhL:b32v57) ([raw file](https://github.com/vaadin/vaadin-context-menu/blob/b7ebc127fbe247dcc4f1ac6364a651bb2658db50/docs/vaadin-context-menu.adoc#L10)):*
<details><summary><i>Previously, tehapo (Teemu Pöntelin) wrote…</i></summary>
> Why not simply mention both options – "which by default is opened by a context click or a long press touch event"?
</details>
+1

---

*[docs/vaadin-context-menu.adoc, line 51 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11#-KRhJG6FVJ55a9dcJWAd:-KS57meFUNr1WX22gRHv:b66p33c) ([raw file](https://github.com/vaadin/vaadin-context-menu/blob/b7ebc127fbe247dcc4f1ac6364a651bb2658db50/docs/vaadin-context-menu.adoc#L51)):*
<details><summary><i>Previously, Saulis (Sauli Tähkäpää) wrote…</i></summary>
> That's right. I'm probably just nitpicking here, don't have a strong opinion about the wording.
</details>
Ok, leave it without the word. The user could figure out  the usage by reading the code. I would change the word `for clicks` by `for events` or by `for the opening event` 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11)*
<!-- Sent from Reviewable.io -->

Saulis commented:
Review status: all files reviewed at latest revision, 8 unresolved discussions.

---

*[docs/vaadin-context-menu.adoc, line 10 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11#-KRhHTIURsuN4_I8nv81:-KS5K37As--2_8RlT_Nl:b-896fix) ([raw file](https://github.com/vaadin/vaadin-context-menu/blob/b7ebc127fbe247dcc4f1ac6364a651bb2658db50/docs/vaadin-context-menu.adoc#L10)):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary>
> +1
</details>
Done.

---

*[docs/vaadin-context-menu.adoc, line 51 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11#-KRhJG6FVJ55a9dcJWAd:-KS5KGDrZcmxERefqohK:b-896fix) ([raw file](https://github.com/vaadin/vaadin-context-menu/blob/b7ebc127fbe247dcc4f1ac6364a651bb2658db50/docs/vaadin-context-menu.adoc#L51)):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary>
> Ok, leave it without the word. The user could figure out  the usage by reading the code. I would change the word `for clicks` by `for events` or by `for the opening event` 
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11)*
<!-- Sent from Reviewable.io -->

tehapo commented:
Reviewed 1 of 1 files at r5.
Review status: all files reviewed at latest revision, 8 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/11)*
<!-- Sent from Reviewable.io -->
